applications: serial_lte_modem: support A-GPS and P-GPS combined (#12359)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/GNSS_AT_commands.rst

   :local:
   :depth: 2

The following list contains GNSS-related AT commands.

Control GNSS
============

The ``#XGPS`` command controls the GNSS.

Set command
-----------

The set command allows you to start and stop the GNSS module.

Both the :ref:`lib_nrf_cloud_agps` service and the :ref:`lib_nrf_cloud_pgps` service can be used with the module, either together or independently of each other.
Using them reduces the time it takes the GNSS module to estimate its position.

To use either of them, the device must be connected to nRF Cloud (using the :ref:`#XNRFCLOUD <SLM_AT_NRFCLOUD>` AT command) when starting the GNSS module.
In addition, the following Kconfig options must be enabled:

   * :kconfig:option:`CONFIG_NRF_CLOUD_AGPS <CONFIG_NRF_CLOUD_AGPS>` to use A-GPS.
   * :kconfig:option:`CONFIG_NRF_CLOUD_PGPS <CONFIG_NRF_CLOUD_PGPS>` to use P-GPS.

> When using either assistive service, the choice of which are used is made at compile time only.

Reviewer: Would this work:
```suggestion
Assistive services cannot be enabled after compile time.
```
Alternatively, if the Kconfig options are what must be enabled at compile time, then it makes sense to move this information to the sentence that introduces them, so using this for Line 26 and removing Line 31:
```
In addition, the following Kconfig options must be enabled at compile time:
```

Reviewer: Hmm, in that case, I might move this to a negative declaration like so:
```suggestion
If both assistive services are enabled at compile time, you cannot choose to use only one of them at run time.
```

* ``0`` - Stop the GNSS module.
  In this case, no other parameter is allowed.
* ``1`` - Start the GNSS module.

``<cloud_assistance>`` is an integer that indicates whether to use the nRF Cloud assistive services that were enabled during compilation.
It is ``0`` for disabled or ``1`` for enabled.

``<interval>`` is an integer that indicates the GNSS fix interval in seconds.
It can have one of the following values:

* ``0`` - Single-fix navigation mode.
* ``1`` - Continuous navigation mode.
  The fix interval is set to 1 second.
* Ranging from ``10`` to ``65535`` - Periodic navigation mode.
  The fix interval is set to the specified value.

In continuous navigation mode, the ``<timeout>`` parameter must not be given.
In single-fix and periodic navigation modes, the ``<timeout>`` parameter indicates the maximum time in seconds that the GNSS receiver is allowed to run while trying to produce a valid PVT estimate.
It can have one of the following integer values:

* ``0`` - The GNSS receiver runs indefinitely until a valid PVT estimate is produced.
* Any positive integer - The GNSS receiver is turned off after the specified time is up, even if a valid PVT estimate was not produced.

> When not specified in single-fix or periodic navigation modes, it defaults to a timeout value of 60 seconds.

Reviewer: ```suggestion
  When not specified in single-fix or periodic navigation modes, it defaults to a timeout value of 60 seconds.
```
Indenting to make it part of the bullet.

Reviewer: Discussed on Teams, and landed on this suggestion:
```suggestion
* No value - In single-fix or periodic navigation modes, the timeout defaults to 60 seconds.
```

   :local:
   :depth: 2

The following list contains GNSS-related AT commands.

Control GNSS
============

The ``#XGPS`` command controls the GNSS.

Set command
-----------

The set command allows to start and stop the GNSS module.

> It can be used with or without :ref:`lib_nrf_cloud_agps` and :ref:`lib_nrf_cloud_pgps`.
> Both assistive services can also be used independently of each other.

Reviewer: Alright, I checked the libraries' own documentation, and they do refer to "nRF Cloud A-GPS service" and "nRF Cloud P-GPS service".
```suggestion
Both the :ref:`lib_nrf_cloud_agps` service and the :ref:`lib_nrf_cloud_pgps` service can be used with the module, either together or independently of each other.
```
